### PR TITLE
tweak some of the HPOS Settings UI

### PR DESCRIPTION
**Pull request by lsinger**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

This PR tweaks a few things on the Settings / Advanced / Features page, especially around the HPOS settings. 

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

0. you will probably need to run `pnpm run --filter=woocommerce/client/legacy build` before being able to observe this PR's changes
1. visit the `/wp-admin/admin.php?page=wc-settings&tab=advanced&section=features` page and observe tweaked wording and less padding between the two HPOS settings
2. observe that the HPOS settings have been moved to the top
3. install and activate a plugin that is incompatible with HPOS and observe the wording and styling of the warning on the `/wp-admin/admin.php?page=wc-settings&tab=advanced&section=features` page

A few of these tweaks are my suggestions, while others have been previously discussed and decided. Happy to adjust. Screenshot of what to expect: 

Screenshot 2023-08-25 at 18 56 46

### Changelog entry

-   [ ] Automatically create a changelog entry from the details below.

<details>

#### Significance

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Comment 

</details>

### CODE REVIEW

File: plugins/woocommerce/src/Internal/DataStores/Orders/CustomOrdersTableController.php [hunk 1]
```diff
@@ -479,16 +479,17 @@ private function get_hpos_setting_for_feature( $sync_status ) {
 
 		return array(
 			'id'          => self::CUSTOM_ORDERS_TABLE_USAGE_ENABLED_OPTION,
-			'title'       => __( 'Data storage for orders', 'woocommerce' ),
+			'title'       => __( 'Order data storage', 'woocommerce' ),
 			'type'        => 'radio',
 			'options'     => array(
 				'no'  => __( 'WordPress post tables', 'woocommerce' ),
```
Comment (jorgeatorres) [inline plugins/woocommerce/src/Internal/DataStores/Orders/CustomOrdersTableController.php]:
```suggestion
				'no'  => __( 'WordPress posts storage (legacy)', 'woocommerce' ),
```

File: plugins/woocommerce/src/Internal/DataStores/Orders/CustomOrdersTableController.php [hunk 2]
```diff
@@ -479,16 +479,17 @@ private function get_hpos_setting_for_feature( $sync_status ) {
...
-				'yes' => __( 'High performance order storage (new)', 'woocommerce' ),
+				'yes' => __( 'High-Performance Order Storage (new)', 'woocommerce' ),
```
Comment (jorgeatorres) [inline plugins/woocommerce/src/Internal/DataStores/Orders/CustomOrdersTableController.php]:
```suggestion
				'yes' => __( 'High-performance order storage (recommended)', 'woocommerce' ),
```

File: plugins/woocommerce/src/Internal/Features/FeaturesController.php [hunk 3]
```diff
@@ -592,6 +594,15 @@ private function add_feature_settings( $settings, $current_section ): array {
...
 
 		$feature_ids              = array_keys( $features );
+		usort( $feature_ids, function( $feature_id_a, $feature_id_b ) use ( $features ) {
+			if ( ! isset( $features[ $feature_id_a ]['order'] ) ) {
+				$features[ $feature_id_a ]['order'] = 0;
+			}
+			if ( ! isset( $features[ $feature_id_b ]['order'] ) ) {
+				$features[ $feature_id_b ]['order'] = 0;
+			}
+			return $features[ $feature_id_b ]['order'] <=> $features[ $feature_id_a ]['order'];
```
Comment (jorgeatorres) [inline plugins/woocommerce/src/Internal/Features/FeaturesController.php]: We can simplify the callback by using the `??` operator, as we wouldn't have to check if `'order'` is set in the `$features` entries nor set it ourselves.
Totally optional, though.

```suggestion
			return ( $features[ $feature_id_b ]['order'] ?? 0 ) <=> ( $features[ $feature_id_a ]['order'] ?? 0 );
```

**Review by jorgeatorres — CHANGES_REQUESTED** [review]
Hi @lsinger!

Great job on this one 💯. Looks much better and I particularly liked the implementation of the `'order'` arg, which is going to be very useful.

I left a few suggestions, but nothing critical. Please let me know what you think.

File: plugins/woocommerce/src/Internal/Features/FeaturesController.php (hunk 3 above)
Comment (lsinger) [reply plugins/woocommerce/src/Internal/Features/FeaturesController.php]: added in [a7614c7](https://github.com/woocommerce/woocommerce/pull/39912/commits/a7614c7e8f08017472ed3f4b21c00cf78b290f27) -- thanks a lot for the suggestion ☺️

**Review by jorgeatorres — APPROVED** [review]
LGTM. Thanks @lsinger!

**Pull request merged**